lib/checks/canFindDenominatorInNumerator.js

```diff
@@ -20,38 +20,69 @@ function canFindDenominatorInNumerator(node) {
   if (Node.Type.isParenthesis(denominator)) {
     denominator = denominator.content;
   }
-  if (!(numerator.op === '+' || numerator.op === '-' ||
-        denominator.op === '+' || numerator.op === '-')) {
+
+  let n_args_length;
+  // If numerator is '*' op, it signifies a single 'ax', should be assigned a
+  // length of 1
+  if ('args' in numerator && numerator.op !== '*') {
+    n_args_length = numerator.args.length;
+  }
+  else {
+    n_args_length = 1;
+  }
+  let d_args_length;
+  if ('args' in denominator) {
+    d_args_length = denominator.args.length;
+  }
+  else {
+    d_args_length = 1;
+  }
+
+  // If numerator isn't length 2 or length 1 with a polynomial return false
+  if (!(n_args_length === 2)) {
+    if (!(n_args_length === 1 || Node.Type.isConstant(numerator))) {
```

Comment:
You're right, it should be a check for polynomial, originally I wanted to put isSymbol() there, but it doesn't work on values that have a coefficient: ex. Node.Type.isSymbol('2x') evaluates to false, so i figured the logical positive would work instead: !Node.Type.isConstant('2x') evaluates to true.

Also, I don't check that the first argument of the numerator/denominator is a polynomial term because I assume that if we've already evaluated through and there's a second arg in the numerator/denominator that's a constant, the only thing left in the first args MUST be a polynomial, cause the other simplifiers would have added like terms already. Do you want me to explicitly write !Node.Type.isConstant(denominator.arg[0])?